rules.py

```diff
@@ -912,6 +912,47 @@ def check(self):
                 "The Election File has following issues with the identifiers.", error_log)
 
 
+class AllCaps(base.BaseRule):
+    """The Name elements in Candidates, Contests and Person elements should not be in all uppercase.
+
+    If the name elements in Candidates, Contests and Person elements are in uppercase, 
+    the list of bojectIds of those elements will be returned to the user as a warning."""
+
+    def elements(self):
+        schema_tree = etree.parse(self.schema_file)
+        required_elements = ["Candidate", "CandidateContest", "Person"]
+        eligible_elements = []
+        for event, element in etree.iterwalk(schema_tree):
+            tag = self.strip_schema_ns(element)
+            if tag and tag == "complexType" and element.get("name") in required_elements:
+                eligible_elements.append(element.get("name"))
+        return eligible_elements
+
+    def check(self, element):
+        object_id = element.get("objectId")
+        if element.tag == "Candidate":
+            ballot_name = element.find("BallotName")
+            if ballot_name.find("Text") is not None:
+                name = ballot_name.find("Text").text
+                if name is not None and name == name.upper():
```

Comment:
Does this check work for various UTF-8 text? E.g.,

NORMA VERÓNICA FLORES PABLO

Reply:
Yes, it works with UTF-8.